services/lambda-pr-status-labeler/PRStatusBot.py

```diff
@@ -0,0 +1,298 @@
+        except KeyError:
+            raise Exception("WebHook from GitHub is not signed")
+        git_signed = git_signed.replace('sha1=', '')
+
+        # Signing our event with the same secret as what we assigned to github event
+        secret = self.webhook_secret
+        body = ast.literal_eval(event["Records"][0]['body'])['body']
+        secret_sign = hmac.new(key=secret.encode('utf-8'), msg=body.encode('utf-8'), digestmod=hashlib.sha1).hexdigest()
+
+        # Validating signatures match
+        return hmac.compare_digest(git_signed, secret_sign)
+
+    def _get_github_object(self):
+        """
+        This method returns github object initialized with Github personal access token
+        """
+        github_obj = Github(self.github_personal_access_token)
+        return github_obj
+
+    def _get_pull_request_object(self, github_obj, pr_number):
+        """
+        This method returns a PullRequest object based on the PR number
+        :param github_obj
+        :param pr_number
+        """
+        repo = github_obj.get_repo(self.repo)
+        pr_obj = repo.get_pull(pr_number)
+        return pr_obj
+
+    def _get_commit_object(self, github_obj, commit_sha):
+        """
+        This method returns a Commit object based on the SHA of the commit
+        :param github_obj
+        :param commit_sha
+        """
+        repo = github_obj.get_repo(self.repo)
+        commit_obj = repo.get_commit(commit_sha)
+        return commit_obj
+
+    def _drop_other_pr_labels(self, pr, desired_label):
+        labels = pr.get_labels()
+
+        for label in labels:
+            logging.info(f'Label:{label}')
+            if label.name.startswith('pr-') and label.name != desired_label:
+                try:
+                    logging.info(f'Removing {label}')
+                    # pr.remove_from_labels(label)
+                except Exception:
+                    logging.error(f'Error while removing the label {label}')
+
+    def _add_label(self, pr, label):
+        # drop other PR labels
+        self._drop_other_pr_labels(pr, label)
+
+        # check if the PR already has the desired label
+        if(self._has_desired_label(pr, label)):
+            logging.info(f'PR {pr.number} already contains the label {label}')
+            return
+
+        # try:
+        #     pr.add_to_labels(label)
+        # except Exception:
+        #     logging.error(f'Unable to add label {label}')
+
+        # verify that label has been correctly added
+        if(self._has_desired_label(pr, label)):
+            logging.info(f'Successfully labeled {label} for PR-{pr.number}')
+        else:
+            logging.info(f'Not labeled {label}')
+        return
+
+    def _has_desired_label(self, pr, desired_label):
+        """
+        This method returns True if desired label found in PR labels
+        """
+        labels = pr.get_labels()
+        for label in labels:
```

Comment:
You can directly check if the label is inside the collection. No need to do that loop

Reply:
Labels is a PaginatedList of objects of type Label. Each object has an attribute `name`
I guess Python doesn't allow to check if objects in a list contain an attribute value without iteration.
I could use 
```
any(label.name == desired_label for label in labels)
```
But in general I'd have to iterate. Correct me if wrong.

Reply:
Calling list(...) On it should iterate the entire paginated list automatically. Then, that operation should work. But I'll leave it up to you :)

Reply:
Sure. I'll leave it as this.